File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRContainer.java
package com.pennapps.labs.pennmobile.classes;

import java.util.ArrayList;

/**
 * Created by MikeD on 2/10/2018.
 */

public class GSRContainer {

    private String gsrName;
    //used to keep track availability of the given room
    private ArrayList<GSRContainerSlot> gsrAvailableSlots = new ArrayList<GSRContainerSlot>();


    public GSRContainer(String constructorGsrName, String constructorTimeRange, String constructorDateTime,
                        String constructorDayDate, String constructorDateNum, String constructorDuration, String constructorElementId) {

        gsrName = constructorGsrName;

        GSRContainerSlot newGSRSlot = new GSRContainerSlot(constructorTimeRange, constructorDateTime, constructorDayDate, constructorDateNum, constructorDuration, constructorElementId);

        gsrAvailableSlots.add(newGSRSlot);

    }

    public void addGSRSlot(String constructorTimeRange, String constructorDateTime,
                           String constructorDayDate, String constructorDateNum, String constructorDuration, String constructorElementId) {

        //created new GSR time slot object
        GSRContainerSlot newGSRSlot = new GSRContainerSlot(constructorTimeRange, constructorDateTime, constructorDayDate, constructorDateNum, constructorDuration, constructorElementId);

        gsrAvailableSlots.add(newGSRSlot);
    }

    public String getGsrName() {
        return gsrName;
    }

    public ArrayList<GSRContainerSlot> getAvailableGSRSlots() {
        return gsrAvailableSlots;
    }
}

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRContainer.kt
package com.pennapps.labs.pennmobile.classes

import org.joda.time.DateTime
import java.util.ArrayList

/**
 * Created by Varun on 10/14/2018.
 */

class GSRContainer(val gsrName: String, constructorTimeRange: String, constructorStartTime: DateTime,
                   constructorElementId: String, val gid: Int, val roomId: Int) {
    //used to keep track availability of the given room
    val availableGSRSlots = ArrayList<GSRContainerSlot>()

    init {

        val newGSRSlot = GSRContainerSlot(constructorTimeRange, constructorStartTime, constructorElementId, gid, roomId, gsrName)

        availableGSRSlots.add(newGSRSlot)

    }

    fun addGSRSlot(constructorTimeRange: String, constructorStartTime: DateTime,
                   constructorElementId: String) {

        //created new GSR time slot object
        val newGSRSlot = GSRContainerSlot(constructorTimeRange, constructorStartTime, constructorElementId, gid, roomId, gsrName)

        availableGSRSlots.add(newGSRSlot)
    }
}

